crates/milli/src/update/new/document_change.rs

```diff
@@ -173,7 +173,7 @@ impl<'doc> Update<'doc> {
     /// Otherwise `false`.
     pub fn has_changed_for_fields<'t, Mapper: FieldIdMapper>(
         &self,
-        fields: Option<&[&str]>,
+        selector: &mut impl FnMut(&str) -> PatternMatch,
```

Comment:
worth updating the documentation of this function?

```rust
    /// Returns whether the updated version of the document is different from the current version for the subset of fields selected by `selector`.
    ///
    /// `true` if at least one top-level-field that is exactly a selected field or a parent of a selected field changed.
    /// Otherwise `false`.
    /// 
    /// - Note: `_geo` and `_vectors` are not taken into account by this function.
```